great_expectations/datasource/fluent/file_path_data_asset.py

```diff
@@ -188,6 +228,102 @@ def build_batch_request(
             partitioner=partitioner,
         )
 
+    def _add_batch_definition_path(self, name: str, path: PathStr) -> BatchDefinition:
+        """Adds a BatchDefinition which matches a single Path.
+
+        Parameters:
+            name: BatchDefinition name
+            path: File path relative to the Asset
+
+        Raises:
+             PathNotFoundError: path cannot be resolved
+             AmbiguousPathError: path matches more than one file
+        """
+        regex = re.compile(str(path))
+        matched_data_references = len(self._data_connector.get_matched_data_references(regex=regex))
+        # we require path to match exactly 1 file
+        if matched_data_references < 1:
+            raise PathNotFoundError(path=path)
+        elif matched_data_references > 1:
+            raise AmbiguousPathError(path=path)
+        return self.add_batch_definition(
+            name=name,
+            partitioner=PartitionerPath(regex=regex),
+        )
+
+    def _add_batch_definition_yearly(self, name: str, regex: re.Pattern) -> BatchDefinition:
+        """Adds a BatchDefinition which defines yearly batches by file name.
+
+        Parameters:
```

Comment:
nit: Google style guide uses `Args` instead of `Parameters`: https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

Reply:
oh great! thanks for flagging - i'll update in my followup PR